sympy/ntheory/ecm.py

```diff
@@ -0,0 +1,245 @@
+#----------------------------------------------------------------------------#
+#                                                                            #
+#                   Lenstra's Elliptic Curve Factorization                   #
+#                                                                            #
+#----------------------------------------------------------------------------#
+
+
+class Point:
+    """For denoting the point on the curve.
+    For ECM we don't need the y-coordinate. All
+    arithmetic operations use only the x and z coordinates.
+    """
+
+    def __init__(self, x, z):
+        self.x = x
+        self.z = z
+
+def _pt_add(P, Q, diff, n):
```

Comment:
I would make this a method of `Point`. For example `def add(P, Q, diff,n):`. It is not necessary to change `P` to `self`.

Reply:
Shall i make them as `@staticmethod`

Reply:
Done.